.github/workflows/cherry-pick.yml

```diff
@@ -0,0 +1,106 @@
+# development branches for each version of Mariner. By default, the workflow is run
+# whenever a CVE fix PR to a fast-track branch is merged. Additional target branches
+# for cherry-pick can be specified by adding a comment to the original PR with the
+# following syntax:
+#   /cherry-pick <target-branch-1> <target-branch-2>
+# Note the list of target branches is space-separated, and the commenter must be a
+# member or owner of the repository for the action to work.
+# After cherry-pick succeeds for a branch, a new PR will be created to merge the commit
+# to that target branch, and a label will be added to the original PR to indicate that
+# the commit has been cherry-picked to that branch.
+name: Cherry pick commits from PR
+on:
+  pull_request:
+    types:
+      - closed
+    branches:
+      - 'fasttrack/*'
+    paths:
+      - 'SPECS/**'
+      - 'SPECS-EXTENDED/**'
+      - 'SPECS-SIGNED/**'
+
+permissions:
+  contents: read
+  pull-requests: read
+defaults:
+  run:
+    shell: bash
+
+env:
+  CHERRY_PICK_BRANCH_MAPPING: '{"fasttrack/1.0": "1.0-dev", "fasttrack/2.0": "main"}'
+
+jobs:
+  # Scans the PR for cherry pick comments, in addition to the default cherry pick target
+  collect_target_branches:
+    if: github.event.pull_request.merged == true
+    runs-on: ubuntu-latest
+    name: Collect target branches from Pull Request
+    outputs:
+      cherry_pick_target_branches: ${{ steps.target_branches.outputs.cherry_pick_target_branches }}
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      # Collect target branches in a list. The list includes:
+      # - default target branch
+      # - any branch specified in PR comments with the syntax
+      #     /cherry-pick <branch-1> <branch-2>
+      # The final list will only contain unique branches
+      - name: Create list of target branches
+        id: target_branches
+        run: |
+          cherry_pick_branches_json=$(gh pr view ${{ github.event.number }} \
+                                       --repo ${{ github.repository }} \
+                                       --json comments \
+                                       --jq '.comments
+                                              | map(select(.authorAssociation == "MEMBER" or .authorAssociation == "OWNER")
+                                                    | .body
+                                                    | select(startswith("/cherry-pick "))
+                                                    | ltrimstr("/cherry-pick ")
+                                                    | split(" "))
+                                              | flatten
+                                              | . += ["${{ fromJSON(env.CHERRY_PICK_BRANCH_MAPPING)[github.base_ref] }}"]
+                                              | unique')
+          branches_list=$(echo $cherry_pick_branches_json | jq -r @sh | tr -d "'")
+          IFS=" " read -ra target_branches <<< "$branches_list"
```

Comment:
Just curious: what is the format of `branches_list` that we need the additional step to convert it into an array only to expand it in the next line? Nothing important, I was just looking for a way to save a step.

Reply:
`branches_list` is a string containing all the target branch names. I need to separate the names in order to iterate over them

Reply:
But in what format? As in: separate by spaces, new lines, it's a JSON, etc.?:) What I was getting at, is if it's already a space-separated list, then I think we can pass it directly to the `echo` below and skip the step where we create the `target_branches` array.

Reply:
It is a space-separated list, e.g `"1.0-dev main"`. If I try to use a for loop to iterate over the tokens of `branches_list`, it won't work, i.e it treats `"1.0-dev main"` as a single item. I don't think flushing to the GITHUB_OUTPUT file would change that :/

Reply:
hmm actually it worked. Bash still confuses me sometimes